### PR TITLE
Added Unit Tests

### PR DESCRIPTION
**Pull request by abidrahman**
Issue #22 

Added unit tests for Habit, HabitEvents, and HabitStatus. Slight modifications to the classes were also made.

### CODE REVIEW

File: app/src/test/java/cmput301f17t13/com/catisadog/HabitTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Test
+    public void setSchedule() throws Exception {
+        DateTime now = new DateTime();
+        HashSet<Integer> schedule = new HashSet<>(1,2);
+        HabitStatus status = new HabitStatus("Good", new DrawableContainer());
+
+        Habit newHabit = new Habit("1", "Test Habit", "Test Reason", now, schedule, status);
+
```
Comment (nwlieb) [inline app/src/test/java/cmput301f17t13/com/catisadog/HabitTest.java]: Does not test Habit.setSchedule (ie setSchedule is not called)

**Comment by jhryniw** [comment]
My only comment is that HabitStatus should have a private constructor and public static members.

```java
public class HabitStatus {
    public static final HabitStatus ON_TRACK = new HabitStatus("On track", new DrawableContainer());

    private String message;
    private Drawable icon; //Resource key

    private HabitStatus(String message, Drawable icon) {
        this.message = message;
        this.icon = icon;
    }

    public Drawable getIcon() {
        return icon;
    }

    public String getMessage() {
        return message;
    }
}
```

File: app/src/test/java/cmput301f17t13/com/catisadog/HabitTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import static org.junit.Assert.*;
+
+/**
+ * Created by abidrahman on 2017-10-22.
+ */
+public class HabitTest {
+
+    @Test
+    public void setTitle() throws Exception {
```
Comment (jhryniw) [inline app/src/test/java/cmput301f17t13/com/catisadog/HabitTest.java]: setTitle should check the less than 20 characters condition

File: app/src/test/java/cmput301f17t13/com/catisadog/HabitTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        HabitStatus status = new HabitStatus("Good", new DrawableContainer());
+
+        Habit newHabit = new Habit("1", "Test Habit", "Test Reason", now, schedule, status);
+
+        newHabit.setTitle("SampleTitle");
+        assertTrue(newHabit.getTitle() == "SampleTitle");
+    }
+
+    @Test
+    public void setReason() throws Exception {
```
Comment (jhryniw) [inline app/src/test/java/cmput301f17t13/com/catisadog/HabitTest.java]: set Reason should check the less than 30 characters condition

File: app/src/test/java/cmput301f17t13/com/catisadog/HabitEventTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        Location location = new Location("TestLocation");
+        HabitEvent newHabitEvent = new HabitEvent("1", newHabit, user);
+        newHabitEvent.setLocation(location);
+
+        assertTrue(newHabitEvent.getLocation() == location);
+    }
+
+    @Test
+    public void setPhotoUrl() throws Exception {
```
Comment (jhryniw) [inline app/src/test/java/cmput301f17t13/com/catisadog/HabitEventTest.java]: Should check the photo size condition (or at least attempt to?)

**Pull request merged**